src/java/org/apache/cassandra/schema/ColumnMetadata.java

```diff
@@ -275,6 +318,11 @@ public boolean isMasked()
         return mask != null;
     }
 
+    public boolean hasConstraint()
+    {
+        return hasConstraints;
```

Comment:
Why cache this instead of always evaluating `!columnConstraints.isEmpty()`?

Reply:
well yeah this might be changed back if it is so simple stupid implementation which just does size == 0

Reply:
Like in a previous comment, I don't mind being explicit with the caching here. If the room feels against it, I can revert it.

Reply:
I'm in the "cache when you've measured that you need to" camp, so I say drop it unless a profiler informs us otherwise.